src/docfx/build/markdown/BuildMarkdown.cs

```diff
@@ -11,20 +10,19 @@ internal static class BuildMarkdown
         public static Task Build(Context context, Document file, TableOfContentsMap tocMap)
         {
             var markdown = file.ReadText();
-            var model = new PageModel<string> { Content = Markup(markdown) };
+            var (html, yamlHeader) = MarkdownUtility.Markup(markdown, file, context);
+            var content = !string.IsNullOrEmpty(html) ? $"<div>{html.Trim()}</div>" : "";
+            var model = new PageModel<string> { Content = content };
 
             context.WriteJson(model, file.OutputPath);
 
             var metadata = Metadata.FetchFromConfig(file);
```

Comment:
>FetchFromConfig [](start = 36, length = 15)

would we cache that? otherwise it will fetch from config for every file

Reply:
*fetch* is misleading here.

Reply:
we can cache the merge metadata from config via file metadata key

---
In reply to: [188828823](https://github.com/dotnet/docfx/pull/2778#discussion_r188828823) [](ancestors = 188828823)

Reply:
Agreed, @superyyrrzz